operators/src/source/ogr_source.rs

```diff
@@ -420,12 +285,15 @@ where
     FeatureCollectionRowBuilder<G>: FeatureCollectionBuilderGeometryHandler<G>,
 {
     type Output = FeatureCollection<G>;
-    fn query(&self, query: QueryRectangle, ctx: QueryContext) -> BoxStream<Result<Self::Output>> {
+    fn query<'a>(
+        &'a self,
+        query: QueryRectangle,
+        ctx: &'a dyn QueryContext,
+    ) -> BoxStream<'a, Result<Self::Output>> {
         OgrSourceStream::new(
-            self.dataset_information.clone(),
-            self.default_geometry.clone(),
+            self.dataset_information.get(query).unwrap(), // TODO: handle error
```

Comment:
new task to return Result<Stream>?

Reply:
https://trello.com/c/Q3jlJKiT/249-refactor-query-processor-trait-to-return-resultboxstream